### PR TITLE
Feat: Manage Reputation via Multi-Sig

### LINKED ISSUE #2190: [Multi-sig] feat: implement reputation smite multi-sig motion
### Story

As a *user moderating a colony*, I want to be able to create and finalize a multi-sig reputation smite motion.

### Description

After the groundwork has been laid with #2150 , implement the reputation smite motion.

### This issue is complete when...

- [ ] The reputation smite multi-sig motion can be created
- [ ] The motion can be finalized
- [ ] Only users with [adequate permissions](https://www.notion.so/colony/Multi-sig-Permissions-6aaf0b95309f4754a243d54e8d9c4189?pvs=4#fec3699a5fd241a2925bd2809494b157) can create them

### PR DESCRIPTION
**Pull request by davecreaser**
## Description

- Allow users to manage reputation using a multi-sig action.

Block Ingestor changes [here](https://github.com/JoinColony/block-ingestor/pull/253)

## Testing

**First**: Restart your dev env so that you have the correct block ingestor version running. Also visit http://0.0.0.0:3001/reputation/monitor/toggle (make sure you do this only once [or three times, or five...]! - this is explained [here](https://www.notion.so/colony/Reputation-in-development-c05b7a1f79fc4465a220c2806792e4a7)) so that reputation updates happen in dev.

* Step 1. As Leela, install Multi Sig Extension, and set the threshold to 2 (then refresh the page).
* Step 2. Create an action to give Amy multi-sig permissions (then refresh the page).
Screenshot 2024-07-11 at 17 28 11

* Step 3. Navigate to the Contributors tab, and make a note of Fry's reputation. Here you can see it is 3.12%.
Screenshot 2024-07-11 at 17 31 22

* Step 4. As Leela, create an action to award Fry some reputation using multi-sig as the decision method.
Screenshot 2024-07-11 at 17 32 15

* Step 5. As Amy (I suggest an incognito window), navigate to the above action, and using the sidebar approve and finalise the motion.
Screenshot 2024-07-11 at 17 32 52

* Step 6. Go back to the Contributors tab from step 3, and refresh the page (you may need to wait a few seconds or refresh a couple of times because of the way the hook / lambda is currently set up to fetch reputation updates). You should see that Fry's reputation has changed reflecting your action.
Screenshot 2024-07-11 at 17 34 13

* Step 7. Repeat steps 4, 5 and 6, however this time try to _remove_ reputation from Fry, instead of awarding it.
Screenshot 2024-07-11 at 17 35 47
Screenshot 2024-07-11 at 17 36 06
Screenshot 2024-07-11 at 17 37 27

* Optional Step 8. Repeat steps 3, 4, 5 and 6, but this time do everything in a sub domain. This means checking the current reputation percentage in the Contributors tab after first changing the current team at the top of the page to a sub domain, then creating the remove reputation action in that sub domain instead of General. Things should still work the same way!

## Diffs

**Changes** 🏗

* Edited the Manage Reputation Motion saga to handle multi sig.

Resolves #2190

### CODE REVIEW

**Review by mmioana — CHANGES_REQUESTED** [review]
Awesome job on this one @davecreaser! 🚀 

I managed to award Fry reputation
Screenshot 2024-07-12 at 10 03 27
Screenshot 2024-07-12 at 10 03 50
Screenshot 2024-07-12 at 10 04 05

However, when trying to remove Fry's reputation I noticed in the action sidebar the current reputation value remains 0. Can you please re-check this?
Screenshot 2024-07-12 at 10 05 14

**Comment by davecreaser** [comment]
> Awesome job on this one @davecreaser! 🚀
> 
> I managed to award Fry reputation Screenshot 2024-07-12 at 10 03 27 Screenshot 2024-07-12 at 10 03 50 Screenshot 2024-07-12 at 10 04 05
> 
> However, when trying to remove Fry's reputation I noticed in the action sidebar the current reputation value remains 0. Can you please re-check this? Screenshot 2024-07-12 at 10 05 14

Hey @mmioana! Thanks for the review. It looks like reputation might be broken in general on your dev env - I see in the screenshots that in the user hub button in the top right it doesn't show any reputation percentage, which means that the reputation likely isn't being loaded properly anywhere. Could you restart your dev env and try again? You'll need to see a percentage at the top right (in the testing instructions first screenshot you can see Leela has 89.08% rep for example). Thanks!

File: src/components/v5/common/ActionSidebar/partials/forms/ManageReputationForm/utils.ts [hunk 1]
```diff
@@ -39,6 +44,9 @@ export const getManageReputationPayload = (
     annotationMessage: description,
     customActionTitle: '',
     motionDomainId: createdIn,
+    isMultiSig,
+    colonyRoles: extractColonyRoles(colony.roles),
+    colonyDomains: extractColonyDomains(colony.domains),
```
[the remark this answers is not in the document]
Comment (davecreaser) [reply src/components/v5/common/ActionSidebar/partials/forms/ManageReputationForm/utils.ts:49]: Oh very nice! I hadn't seen that. I'll keep an eye on that PR and update this when it's merged. Nice one! 🎉

**Review by mmioana — APPROVED** [review]
After fixing my environment, @davecreaser I can confirm awarding and removing reputation works as expected 🎉 

Starting reputation of fry
Screenshot 2024-07-12 at 13 44 27

Awarding Reputation
Screenshot 2024-07-12 at 13 47 24
Screenshot 2024-07-12 at 13 47 50
Screenshot 2024-07-12 at 13 50 27

Removing reputation
Screenshot 2024-07-12 at 13 50 10

**Review by bassgeta — CHANGES_REQUESTED** [review]
Awarding via MS works:
image
Removing:
image
Removal in a subdomain also works:
image

However, when creating a motion via voting reputation, the sidebar doesn't show up correctly
image
but it's a bug on `master`, though it's easily fixable by adding these 2 lines to `CompletedAction.tsx`
image

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Nice work on this Dave! I was able to award and remove reputation via multi-sig:

Screenshot 2024-07-15 at 12 41 45

And can confirm the motions are correctly showing the sidebar now:

Screenshot 2024-07-15 at 12 43 21
Screenshot 2024-07-15 at 12 43 49

The only thing missing is the multi-sig meatball on the completed action screen.

Screenshot 2024-07-15 at 12 42 20

Should be:
Screenshot 2024-07-15 at 12 45 11

**Review by iamsamgibbs — APPROVED** [review]
Ignore my previous comments regarding the meatball menu, it is no longer required.

**Review by bassgeta — APPROVED** [review]
Nice solution, awarding and removing is still working for me:
image

The sidebar for voting reputation does show up correctly now (I failed it, not an error on this branch :D ):
image

Nice one :ship: !

**Pull request merged**